x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/__jest__/http_requests.helpers.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import sinon, { SinonFakeServer } from 'sinon';
+import axios from 'axios';
+import axiosXhrAdapter from 'axios/lib/adapters/xhr';
+
+/* eslint-disable @kbn/eslint/no-restricted-paths */
+import { usageCollectionPluginMock } from 'src/plugins/usage_collection/public/mocks';
+
+import { uiMetricService, apiService } from '../../../services';
+
+type HttpResponse = Record<string, any> | any[];
+
+const mockHttpClient = axios.create({ adapter: axiosXhrAdapter });
+
+// Register helpers to mock HTTP Requests
+const registerHttpRequestMockHelpers = (server: SinonFakeServer) => {
+  const setSimulatePipelineResponse = (response?: HttpResponse, error?: any) => {
+    const status = error ? error.status || 400 : 200;
+    const body = error ? JSON.stringify(error.body) : JSON.stringify(response);
+
+    server.respondWith('POST', '/api/ingest_pipelines/simulate', [
+      status,
+      { 'Content-Type': 'application/json' },
+      body,
+    ]);
+  };
+
+  return {
+    setSimulatePipelineResponse,
+  };
+};
+
+export const initHttpRequests = () => {
+  const server = sinon.fakeServer.create();
+
+  // Initialize mock services
+  uiMetricService.setup(usageCollectionPluginMock.createSetupContract());
+  // @ts-ignore
+  apiService.setup(mockHttpClient, uiMetricService);
```

Comment:
It is strange to have this `setup` call and the one above inside the `initHttpRequests` as it is not related.
In other places we have a `setupEnvironement()` where we setup the services and call the `initHttpRequests`. Do you mind keeping it consistent?